jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1287,7 +1287,13 @@ else if (_endOfContent == EndOfContent.UNKNOWN_CONTENT)
                         break;
                     }
 
-                    throw new IllegalCharacterException(_state,b,buffer);
+                    //Ignore all invalid characters
+                    if (complianceViolation(RFC7230,"https://tools.ietf.org/html/rfc7230#section-3.2 Invalid token in header name"))
```

Comment:
I don't think this is too forgiving.  We really only want to allow optional white space... in fact probably only spaces.   I think to be safe we are going to need another state  OWS_AFTER_NAME

Reply:
Also, strictly speaking the compliance mode should be RFC2616 as the OWS was not allowed there, so it would only be accepted in LEGACY mode.

Eventually we will replace this mechanism with some more fine grained.

Reply:
As I understood it correctly RFC2616 is compliant to HTTP 0.9 isn't it? At least the comments state it [HttpCompliance.java](https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-http/src/main/java/org/eclipse/jetty/http/HttpCompliance.java)

Reply:
Yes it is, but HTTP/0.9 has no headers so accepting HTTP/0.9 does not give grace for OWS after a header name.    That is a LEGACY mode - ie something that Jetty used to support but no longer does if it is either in strict RFC2616 or RFC7230 modes

Reply:
But does the Legacy mode distinct between headers written in different cases? Then the Legacy mode is not what we are looking for...

Reply:
And there is the Rub with the current big compliance mode switch!  It's all or nothing, when we really need to ability to turn on an off individual aspects of compliance.  We plan to do this for Jetty-10, but it is too invasive to do in a 9.4 dot release.

Reply:
Okay I got it, then I will add the behavior on the legacy mode for now. Is there any documentation what will behave differently when we activate the Legacy mode per default? There are some more checks in the sources and I didn't understand them all for now...